frontend/packages/console-shared/src/hooks/__tests__/usePinnedResources.spec.ts

```diff
@@ -34,11 +37,16 @@ describe('usePinnedResources', () => {
       },
     ]);
 
+    useModelFinderMock.mockReturnValue({
+      findModel: () => ({ kind: 'Deployment' }),
+    });
     useUserSettingsCompatibilityMock.mockClear();
     setPinnedResourcesMock.mockClear();
   });
 
   it('returns an empty array if user settings are not loaded yet', async () => {
+    window.SERVER_FLAGS.perspectives =
+      '[{ "id" : "dev", "visibility": {"state" : "Enabled" }, "pinnedResources" : [{"version" : "v1", "resource" : "deployments"}]}]';
```

Comment:
I would recommend to start with some test cases that use no customization.

```suggestion
    window.SERVER_FLAGS.perspectives = '';
```